File: Back end/cinemoon/src/main/java/com/proyecto_daw2/cinemoon/Model/Boleto.java
package com.proyecto_daw2.cinemoon.Model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "Boleto")
@Data
public class Boleto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id_boleto;
    private int id_funcion;
    private int id_usuario;
    private int cantidad;
    private double Total;

     @ManyToOne
     @JoinColumn(name = "id_funcion", insertable = false, updatable = false)
     private Funcion funcion;
     
     @ManyToOne
     @JoinColumn(name = "id_usuario", insertable = false, updatable = false)
     private Usuario usuario;
}

File: Front end/cinemoon/src/app/model/boleto.ts
import { Funcion } from "./funcion";
import { Usuario } from "./usuario";

export interface Boleto {
    id_boleto: number;
    id_funcion: number;
    id_usuario: number;
    cantidad: number;
    Total: number;

    funcion: Funcion;
    objUsuario: Usuario;
}
